frontend/src/scenes/insights/insightVizDataLogic.ts

```diff
@@ -329,6 +329,10 @@ export const insightVizDataLogic = kea<insightVizDataLogicType>([
         validationError: [
             (s) => [s.insightDataError],
             (insightDataError): string | null => {
+                // Handle Async Queries here
+                if (insightDataError?.data?.error_message) {
```

Comment:
Quick question: does this only apply to async HogQL queries, or HogQL queries overall? Because if the shape of `insightDataError` is different for async ones specifically, that sounds like a minefield

Reply:
InsightDataError is different for sync and async. See below:

Sync
<img width="700" alt="image" src="https://github.com/PostHog/posthog/assets/1855120/75946fa3-05da-4fb3-9755-d6bea1730317">


Async
<img width="702" alt="image" src="https://github.com/PostHog/posthog/assets/1855120/6d37e92f-e290-4799-9c4b-e415bb0f58ea">

The "status" field does work correctly, but the error message is in a different place.

Reply:
Ah, so they're pretty significantly different. That's a bit annyoing. But let's maybe keep as is for now, and only fix the problem at hand.